components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -100,12 +113,74 @@ public ClientManager() throws WebSubAdapterException {
                     config.getSocketTimeout() + ", maxConnections=" +
                     syncConnectionManager.getMaxTotal() + ", maxConnectionsPerRoute=" +
                     syncConnectionManager.getDefaultMaxPerRoute());
+
+            // Initialize MTLS HttpClient
+            if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                    .isMtlsEnabled()) {
+                mtlsHttpClient = getMTLSClient();
+            }
         } catch (IOException e) {
             throw WebSubHubAdapterUtil.handleServerException(
                     WebSubHubAdapterConstants.ErrorMessages.ERROR_GETTING_ASYNC_CLIENT, e);
         }
     }
 
+    private CloseableHttpClient getMTLSClient() throws WebSubAdapterException {
+
+        try {
+            IdentityKeyStoreResolver resolver = IdentityKeyStoreResolver.getInstance();
+
+            // Load custom keystore and truststore
+            KeyStore customKeyStore = resolver.getCustomKeyStore(SUPER_TENANT_DOMAIN_NAME, WEBSUBHUB_KEYSTORE_NAME);
+            LOG.info("Custom keystore loaded successfully for tenant: " + SUPER_TENANT_DOMAIN_NAME +
```

Comment:
Let's not add info logs. Move all of these a debug logs. else currently if these are added for troubleshooting add a todo